imagery/i.evapo.time/main.c

```diff
@@ -316,7 +316,7 @@ int main(int argc, char *argv[])
 		}	
 	    }
 
-	    sum[MAXFILES] = 0.0;
+	    sum[MAXFILES-1] = 0.0;
 	    for (i = 1; i <= nfiles1; i++) 
```

Comment:
There is more off-by-1 indexing in this module. Do you want to fix it as well?

Reply:
You’re right, we have another index bug there! Bigger than the warning.

Reply:
Looking into this a bit more and testing the indexing with sprinkling in some G_debug(), the indexing seems to me to be fine. But I come to the belief that line `sum[MAXFILES] = 0.0` is not necessary at all. That line of code never contributed in any meaningful way.
It's an attempt to zero out an initial value, which will be done anyway in row 334 (`sum[i]=0.0;`) if reached number of `MAXFILES`.

Reply:
The indexing seems to be wrong to me. E.g. there is
```
    int infd[MAXFILES], infd1[MAXFILES], infd2[MAXFILES];
```
but in L149ff nfiles starts with 1 going to MAXFILES whereas allowed indices for infd and inrast are [0, MAXFILES - 1]
Also in L213 is `double sum[MAXFILES];`, largest allowed index is thus `MAXFILES - 1` and L319 `sum[MAXFILES] = 0.0;` is out of array bounds.
In this module, all `for` loops seem to use invalid indices.

Reply:
Thanks! Although I did test the loops and it seemed to look ok for me, I must admit I never tested reaching end of array (`MAXFILE`). I'll look into this again and try come up with a solution...

Reply:
Now I updated with new indexing. I believe I got it right, I need to some more actual testing (before and after output). Meanwhile, if you have the time please do take a look.

I'm not quite sure though, about the line, from where this all started: `sum[MAXFILES] = 0.0;`, exactly what was intended to be zeroed out.